File: pegasusX/apps/retailer-app-android/app/src/main/java/com/pegasusx/retailer/data/api/RetailerSessionReconcile.kt
package com.pegasusx.retailer.data.api

/** Refetch server-authoritative retailer snapshots after transport reconnect. */
suspend fun reconcileRetailerSession(api: PegasusApi) {
    runCatching { api.getActiveFulfillments() }
    runCatching { api.getPendingPayments() }
    runCatching { api.getTrackingOrders() }
}

File: pegasusX/apps/retailer-app-ios/retailerapp/retailerapp/Services/RetailerSessionReconcile.swift
//
//  RetailerSessionReconcile.swift
//  retailerapp
//
//  Refetch server-authoritative retailer snapshots after transport reconnect.
//

import Foundation

enum RetailerSessionReconcile {
    static func run(api: APIClient = .shared) async {
        _ = try? await api.getActiveFulfillments()
        _ = try? await api.getPendingPayments()
        _ = try? await api.getTrackingOrders()
    }
}
